src/gleam_stdlib.mjs

```diff
@@ -196,8 +196,20 @@ export function split(xs, pattern) {
   return List.fromArray(xs.split(pattern));
 }
 
-export function join(xs) {
-  return xs.toArray().join("");
+export function join(xs, separator) {
+  let result = "";
+  for (const x of xs) {
+    result = result + x + separator;
+  }
+  return result.slice(0, -separator.length);
```

Comment:
Is this faster if we omit the sep for the first one rather than remove a portion of the string afterwards? My (probably wrong) understanding of strings in V8 is that removing part of one is a full copy.

Reply:
```gleam
    let bench_timer = create_timer()
    bench_data
    |> string.join("\n")
    let timing = read_timer(bench_timer)
    io.print(int.to_string(timing) <> "ms\n")
```

Results

```javascript
export function join_(xs, separator) {
  let result = "";
  for (const x of xs) {
    result = result + x + separator;
  }
  return result.slice(0, -separator.length);
}
```
2274ms

```javascript
export function join(xs, separator) {
  let result = "";
  let i = 0;
  for (const x of xs) {
    if (i == 0) {
      result = x
    } else {
      result = result + separator + x;
    }
    i++;
  }
  return result;
}
```
1883ms